lib/src/DynamicProto.ts

```diff
@@ -18,6 +18,22 @@ interface DynamicGlobalSettings {
 
 
 
+/**
+ * Helper to check if we're running in a server-side rendering environment
+ * like Node.js or Cloudflare Workers
+ * @ignore
+ */
+function _isServerSideRender(): boolean {
+    // Check for common server-side environments
+    // 1. Missing window or document (Node.js, some SSR frameworks)
+    // 2. Cloudflare Worker specific environment
+    return (!hasWindow() || 
+        !hasDocument() || 
+        (!!getNavigator() && 
+         getNavigator().userAgent && 
+         getNavigator().userAgent.indexOf('Cloudflare-Workers') >= 0));
```

Comment:
I can't find any documentation that calls out that Cloudflare uses or updates the useragent like this to detect that it's running in a SSR runtime, where is the documentation?

Also as this code is designed to run in node and worker environments it can't just assume that if there is not window or document that it should disable the instance function assignment

Reply:
continue